scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -471,16 +471,12 @@
                                         :cook-expected-state/running
                                         (case pod-synthesized-state-modified
                                           ; This indicates that something deleted it behind our back
-                                          :missing (do
-                                                     (log/info "In compute cluster" name ", something deleted"
-                                                               pod-name "behind our back")
-                                                     ; A :cook-expected-state/running job suddenly disappearing in k8s is
-                                                     ; a sign of a preemption, so treat this case as a missed preemption.
-                                                     ; TODO: When we have a better story for preemption, we should switch this to
-                                                     ; go through the handle-pod-completed stack. That needs to be guarded with a null
-                                                     ; check because handle-pod-completed cannot handle null pods. So, something like:
-                                                     ; (if pod (handle-pod-completed ...) (....write a unknown reason to the pod....))
```

Comment:
Do we not need the `null`-check described here?

Reply:
@scrosby we agreed that we don't need it. can you confirm?

Reply:
We need the null check. When we know its preempted, we don't need the null check; we couldn't have been able to set that flag with a null Pod --- But that's the first branch of the 'if' that doesn't need a null check. We still do here.

Reply:
ok, so what does "write a unknown reason to the pod" look like

Reply:
Roughly the same code as handle-pod-preemption, but with a different reason code.

Reply:
updated